src/client.js

```diff
@@ -13,133 +14,385 @@ export function getAddress(system) {
+                    if (options.command === "response")
+                        imageLoadChannelResponse = options;
+                });
+                await imageLoadChannel.wait();
+                const imageReadChannel = cockpit.channel({
+                    payload: "fsread1",
+                    path: imageFile,
+                    max_read_size: 107374182400,
+                    redirect: imageLoadChannel.id,
+                    binary: true,
+                    host: targetHost
+                });
+                await imageReadChannel.wait();
+                await new Promise(resolve =>
+                    imageReadChannel.addEventListener("control", function (e, options) {
+                        if (options.command === "done")
+                            resolve();
+                    })
+                );
+                imageLoadChannel.control({ command: "done" });
+                await new Promise((resolve, reject) =>
+                    imageLoadChannel.addEventListener("close", function (e, options) {
+                        if (options.problem)
+                            reject(new Error(options.problem));
+                        else if (imageLoadChannelResponse.status >= 200 && imageLoadChannelResponse.status <= 299)
+                            resolve();
+                        else
+                            reject(new Error(String.fromCharCode.apply(null, imageLoadChannelData)));
+                    })
+                );
+            } finally {
+                await cockpit.script(`rm "${imageFile}"`, { host: targetHost });
+            }
+
+            // Inspect the old image and try to tag the new one to match it
+            callbacks.imageTagging();
+            const oldImageInfo = await inspectImage(system, containerInfo.Image, host);
+            for (const tag of oldImageInfo.RepoTags) {
+                const i = tag.lastIndexOf(":");
+                try {
+                    await tagImage(system, containerInfo.Image, tag.substring(0, i), tag.substring(i + 1, tag.length),
+                                   targetHost);
+                } catch (error) {
+                    // Ignore
+                }
+            }
+        } else {
+            throw error;
+        }
+    }
+
+    let containerCreated = false;
+    try {
+        callbacks.containerInspect();
+        await inspectContainer(system, containerInfo.Name, targetHost);
+    } catch (error) {
+        if (error.status === 404) {
+            callbacks.containerCreate();
+            const dummyConfig = {
+                name: containerInfo.Name,
+                image: containerInfo.Image,
+                terminal: true,
+                command: containerInfo.Args
+            };
+            await createContainer(system, dummyConfig, targetHost);
+            containerCreated = true;
+        } else {
+            throw error;
+        }
+    }
```

Comment:
So we need container with the same name to exist on the target host? What happens if it already exits? Can you please also add comment above this section what it does?

Reply:
If the container exists the checkpoint is restored into it. If it doesn't exist, it is created first.

Reply:
> If the container exists the checkpoint is restored into it. 

Does that change the container? I guess we should notify that such container exists on the host and we are fine with changing it to avoid unexpected results.

Reply:
> Does that change the container? I guess we should notify that such container exists on the host and we are fine with changing it to avoid unexpected results.

That depends on the migration settings. If "Do not migrate root file-system changes" is checked, it won't change the container's contents, only restore its state (if the container is running, it fails). I'm not sure about the behavior when the option is unchecked and there are conflicting filesystem changes between the containers.

As of the behavior of the migration UI - perhaps a better solution would be to let the user choose the target container using a select (only valid containers, i.e. ones that are not running and use the same image, would be shown), with an option for creating a new container.